subsystems/provisioning/networkmanager.go

```diff
@@ -691,6 +692,16 @@ func (w *Provisioning) mainLoop(ctx context.Context) {
 
 		if pMode {
 			// complex logic, so wasting some variables for readability
+			deviceRebootAfterOfflineDuration := time.Duration(w.cfg.DeviceRebootAfterOfflineMinutes) * time.Minute
+
+			if w.cfg.DeviceRebootAfterOfflineMinutes > 0 && now.After(lastOnline.Add(deviceRebootAfterOfflineDuration)) {
+				w.logger.Infof("device has been offline for more than %s minutes, rebooting", deviceRebootAfterOfflineDuration)
+				syscall.Sync()
+				err := syscall.Reboot(syscall.LINUX_REBOOT_CMD_RESTART)
```

Comment:
This feels very dangerous. This feels like it'd be a rugpull to running processes. Equivalent to `SYSREQ+b` Does it actually do a normal shutdown? If not, maybe exec "systemctl reboot" to guarantee a normal restart.

Reply:
I believe is the same according to the docs, underhood systemctl is running reboot but will update to use systemctl for consistency 

https://www.man7.org/linux/man-pages/man1/systemctl.1.html